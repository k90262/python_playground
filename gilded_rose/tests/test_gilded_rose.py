# -*- coding: utf-8 -*-
import unittest

import approvaltests

from gilded_rose import Item, GildedRose


def print_item(item):
    return f"Item({item.name}, sell_in={item.sell_in}, quality={item.quality})"


def test_update_quality():
    item = Item("foo", 0, 0)
    items = [item]
    gilded_rose = GildedRose(items)

    gilded_rose.update_quality()

    approvaltests.verify(print_item(item))


