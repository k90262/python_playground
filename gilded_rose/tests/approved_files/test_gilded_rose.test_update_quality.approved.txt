Item(foo, sell_in=-1, quality=0)
